packages/backend/src/apps/postman/actions/send-transactional-email/index.ts

```diff
@@ -71,21 +69,40 @@ const action: IRawAction = {
       )
     }
 
+    /**
+     * TODO: properly handle attachment saving
+     */
     const attachmentFiles = await Promise.all(
       result.data.attachments?.map(async (attachment) => {
         const obj = await getObjectFromS3Id(attachment)
         return { fileName: obj.name, data: obj.data }
       }),
     )
 
-    const { recipients, newProgress } = await getRatelimitedRecipientList(
-      result.data.destinationEmail,
-      +progress,
-    )
+    let recipientsToSend = result.data.destinationEmail
+    /**
+     * Logic to handle retries here:
+     * We dont want to send the email to the same recipient again if it has been sent before
+     */
+    const lastExecutionStep = await $.getLastExecutionStep({
+      sameExecution: true,
+    })
+    const isPartialRetry =
+      lastExecutionStep &&
+      lastExecutionStep.status === 'success' &&
+      lastExecutionStep.errorDetails
 
-    let results
-    try {
-      results = await sendTransactionalEmails($.http, recipients, {
+    const oldDataOut =
+      lastExecutionStep?.dataOut as unknown as SendTrasactionEmailDataOut
+    if (isPartialRetry) {
+      const { status, recipient } = oldDataOut
```

Comment:
as discussed - probably can re-use `oldDataOut` as long as recipients list is non-empty. Should not be possible for there to be new `recipientsToSend` compared to `oldDataOut.recipients`